Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -196,11 +233,15 @@ void TRDGlobalTracking::endOfStream(EndOfStreamContext& ec)
        mTimer.CpuTime(), mTimer.RealTime(), mTimer.Counter() - 1);
 }
 
-DataProcessorSpec getTRDGlobalTrackingSpec(bool useMC, bool useTrkltTransf)
+DataProcessorSpec getTRDGlobalTrackingSpec(bool useMC, bool useTrkltTransf, o2::dataformats::GlobalTrackID::mask_t src)
 {
```

Comment:
I think the name assigned to the the device should depend on the input: in the async workflow we will have both instances: using TPC tracks and TPC-ITS and/or global. Could you simply concatenate to "trd-globaltracking" the inputs, e.g.
`o2::utils::concat_string("trd-globaltracking",o2::dataformats::GlobalTrackID::getSourcesNames(src))`

For the same reason, we better define different output names, e.g.
````
outputs.emplace_back(o2::header::gDataOriginTRD,  o2::dataformats::GlobalTrackID::includesDet(DetID::ITS) ?  "MATCHTRD_GLO" : "MATCHTRD_TPC", 0, Lifetime::Timeframe);
````

Reply:
Hi @shahor02 is there an advantage of having on instance of the TRD tracker with both inputs (ITS-TPC and TPC) simultaneously? Because technically it is currently possible and if I would allow that than the output name would need to be defined differently, as there could be both `MATHTRD_GLO` and `MATCHTRD_TPC` from the same device. Or if not should I add an error in case both inputs are defined to avoid that mixture?

Reply:
I meant something else. The idea is to run 1st TPC-TRD matching, which will produce e.g. `MATCHTRD_TPC`.
After similar step in TOF, e.g. inputs TPC and TPC-TRD, the matching to ITS is done for unambigous matches in inputs TPC-TRD-TOF, TPC-TRD, TPC-TOF and TPC, producing either the 4 outputs with ITS added or single output with all ITS-matched tracks mixed (still have to decided which is better, but I have slight preference for single output). Then these outputs should go again to TRD matcher which would match ITS-TPC part of tracks, and,
1) if needed, disable existing TPC-TRD matches not used by ITS-TPC but whose TPC tracks were matched to ITS (in case a greedy matching was done initially for TPC-TRD). By disabling I mean adding approved TPC-TRD tracks to the same output as newly matched ITS-TPC-TRD, so that later the "bad ones" can be filtered out by the RecoContainer logic. 
 2) if initial matching was allowing only unambiguous matches, which were all used in ITS matching as seeds, then this incremental TRD matching should match the rest to improve the eff.

If this is confusing, we can chat on Wed.